feat: handle Chicago data

BumbleFlash opened the pull request:
This PR handles the new Chicago open TNC data (#3).

Tasks:
- [x] input filename as a command-line argument (#4)
- [x] create Chicago model data class and its respective utils class
- [x] display all the data in the terminal  
- [x] Update readme

BumbleFlash committed:
chore: added command line argument feature.

BumbleFlash committed:
chore: created chicago data model class

BumbleFlash committed:
chore: created utils class for the ChicagoData.kt

BumbleFlash committed:
chore: extracted information from ChicagoData and printed the same on the terminal

BumbleFlash committed:
docs: updated README.md

BumbleFlash commented:
@barbeau Quick question, should I upload the Chicago data file in the repo or should we keep it private?

barbeau commented on README.md:
```diff
@@ -14,9 +14,9 @@ To build the application use `mvn clean package` command. This command will crea
 
 ## Run
-To run the application use `java -jar` command. 
+To run the application use `java -jar` command. Make sure to enter the filename as an argument
```
Add note about processing multiple file names

barbeau replied:
Above in line 9-10 add note on Chicago data and link to data

barbeau commented on src/main/java/edu/usf/cutr/grha/utils/ChicagoDataUtils.kt:
```diff
@@ -0,0 +1,16 @@
+package edu.usf.cutr.grha.utils
```
Please also add small set of sample data

barbeau commented on src/main/java/edu/usf/cutr/grha/parsing/CsvBeanFileParser.java:
```diff
@@ -5,9 +5,12 @@
+import java.io.Reader;
 import java.util.List;
 
 public class CsvBeanFileParser extends ProcessorMain {
```
Is there a reason why this `extends ProcessorMain`? I would think it would be it's own class, not extending anything.

I'd also recommend splitting each file type parsing into it's own parser. So you could have `ChicagoDataParser`, `GPSTestParser`, etc.

BumbleFlash replied:
I was extending the `ProcessorMain` to make use of the `getReader()` declared in the class, looked neater than calling `ProcessorMain.getReader()`.

barbeau commented on src/main/java/edu/usf/cutr/grha/utils/ChicagoDataUtils.kt:
```diff
@@ -0,0 +1,16 @@
+object ChicagoDataUtils {
+    @JvmStatic
+    fun printChicagoData(data: List<ChicagoData>) {
+        for (d in data) {
```
I think you could just print `data.toString()` here instead of printing it all our yourself?

BumbleFlash replied:
I would but printing this way, it looks much nicer and easier to read on the terminal. If I use `toString()`, it goes really wide on the terminal for bigger datasets like Chicago Data.

barbeau reviewed (changes requested):
Thanks @BumbleFlash! Some comments below.

BumbleFlash committed:
docs: updated README.md to reflect multiple filenames arguments and new Chicago data file

BumbleFlash committed:
chore: added a utility class that splits the two file names (mandating chicago data to be entered first) and created two different parsers for the data

BumbleFlash committed:
chore: added reduced chicago data

barbeau committed:
refactor: Restructure parsers to extend a common base class

Various other code changes to simplify implementation and allow the parsed list of objects to be returned from the parsers. This allows them to be printed or used in unit tests.

Also modify the command-line parameters to assume first is GPSTest file and second is Chicago TNC CSV file.

barbeau commented:
@BumbleFlash I pushed some changes in https://github.com/CUTR-at-USF/gnss-ride-hailing-analyzer/pull/5/commits/1fe2dc521e5a5de1825e34d02ce10107f7203844 that are more aligned with what I was thinking. This implementation has a common parser base class with the different implementations extending that to get the common code you were previously storing in the Main file. I think this follows OOP principles better.

I've also simplified the command-line parameter code to just assume the first file is GNSS data and the 2nd is the Chicago data.

Please take a look and let me know what you think - you can merge if everything works and looks good to you.

BumbleFlash committed:
chore: fixed typo

BumbleFlash commented:
@BumbleFlash Thank you for working on this. Now, I see what you meant. Looks great to me!
